ipq40xx: add e2600ac c1/c2 to DSA

zhangsd20 opened the pull request:
1. Convert E2600ac c1/c2 to DSA and enable it.
2. Modify the kernel size(E2600AC c1) to fit the new kernel-5


Signed-off-by: 张鹏 <sd20@qxwlan.com>

Ansuel commented on target/linux/ipq40xx/base-files/etc/board.d/02_network:
```diff
@@ -90,6 +90,12 @@ ipq40xx_setup_interfaces()
+		ucidef_set_interfaces_lan_wan "lan1" "eth1"
+		;;
+	qxwlan,e2600ac-c2)
+		ucidef_set_interfaces_lan_wan "lan1 lan2" "eth1"
```
eth1? is this correct?

Ansuel commented:
Please add a commit description of what you are doing.

Also please split this in 3 commits, one for enlarging the kernel size and 2for enabling the device and the dsa configuration for each one...

Ansuel commented:
(you can have multiple commit in one pr)

zhangsd20 commented:
Ok,thanks.


Sorry I'm trying how to do it. I haven't done anything like this before











在2023年02月22 09时20分，"Christian ***@***.***>写道：


(you can have multiple commit in one pr)
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

zhangsd20 committed:
ipq40xx: enlarging Qxwlan E2600AC c1 kernel size

kernel-5.15 size greater than 4096k.so enlarging kernel size 4096k to 5120k.

Signed-off-by: 张 鹏 <sd20@qxwlan.com>

zhangsd20 committed:
ipq40xx: add e2600ac c1 dsa

Convert E2600ac c1 to DSA and enable it.

Signed-off-by: 张 鹏 <sd20@qxwlan.com>

zhangsd20 committed:
ipq40xx: add e2600ac c2 dsa

Convert E2600ac c2 to DSA and enable it.

Signed-off-by: 张 鹏 <sd20@qxwlan.com>

Ansuel commented:
@zhangsd20 Hi, why you are closing this? It would be good to continue things here.

Anyway commit should be imperative.
I would ask you to change the first commit to

`ipq40xx: enlarge Qxwlan E2600AC c1 kernel size to 5120k`

and as commit description

```
Kernel 5.15 needs more space as the kernel is now bigger than the current limit.
Enlarge kernel size from 4096k to 5120k.
```

With this change this is ready and i will merge it.

zhangsd20 commented:
I'm sorry. Since I saw that the author's name did not match Signed-off-by,

I wanted to fix this issue. But when operating locally, the wrong operation was carried out,

resulting in some problems. I can only recreate a PR











在2023年02月22 21时57分，"Christian ***@***.***>写道：


@zhangsd20 Hi, why you are closing this? It would be good to continue things here.
Anyway commit should be imperative.
I would ask you to change the first commit to
ipq40xx: enlarge Qxwlan E2600AC c1 kernel size to 5120k
and as commit description
Kernel 5.15 needs more space as the kernel is now bigger than the current limit.
Enlarge kernel size from 4096k to 5120k.With this change this is ready and i will merge it.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

zhangsd20 commented:
If this PR can be merged, then I will remove the new PR.

Thank you very much!

zhangsd20 commented:
If this PR can be merged, then I will remove the new PR.

Thank you very much!

robimarko commented:
@Ansuel I posted comments in the new PR, this is just confusing.

So, to make the same questions available here:
1. What is the bootloader limit for kernel size, or is there even one?
Cause, if there is none there is no point in defining the kernel limit, and if there is one just set it now and not update every time the building breaks.

2. eth1 as the name makes no sense, is it printed on the device or?
If not, then make it `wan` or `lan2`

zhangsd20 committed:
ipq40xx: add support e2600ac c1 kernel size dynamic splitter

Then remove the kernel size completely with that explanation as a dynamic splitter is being used anyway.

Signed-off-by: 张 鹏 <sd20@qxwlan.com>

robimarko commented:
You dont need the commit enlarging kernel size at all.

Also, the name and description for the `KERNEL_SIZE` removal are not really great.
It should probably be more like this:
```
ipq40xx: e2600ac-c1: remove KERNEL_SIZE

Currently, e2600ac-c1 cannot be built as the kernel is larger than the defined KERNEL_SIZE,
however, there is no bootloader limit for the kernel size so remove KERNEL_SIZE completely.
```

zhangsd20 commented:
I've tested it, it is fine.
I added a new commit as you suggested











在2023年02月23 19时55分，"Robert ***@***.***>写道：


You dont need the commit enlarging kernel size at all.
Also, the name and description for the KERNEL_SIZE removal are not really great.
It should probably be more like this:
ipq40xx: e2600ac-c1: remove KERNEL_SIZE

Currently, e2600ac-c1 cannot be built as the kernel is larger than the defined KERNEL_SIZE,
however, there is no bootloader limit for the kernel size so remove KERNEL_SIZE completely.—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you modified the open/close state.Message ID: ***@***.***>
